### ISSUE TITLE
Folder structure is quite messy

### ISSUE THREAD

**Issue by sleepylee** [rahatshahzeb/N3#4]
Activity, fragment, base classes, application, CircleIndicator...all in 1 place, shouldn't we manage it like the other place rather than half-baked?

**Comment by rahatshahzeb** [comment]
I agree with you but since this was a coding challenge and my focus was on code quality, I kept it simple.

**Comment by sleepylee** [comment]
Code structure is also considered as a standard in code quality I believe. Bad structure gives bad experience to the first approach.
